Content.Server/Anomaly/AnomalySystem.Generator.cs

```diff
@@ -128,6 +133,22 @@ public void SpawnOnRandomGridLocation(EntityUid grid, string toSpawn)
             if (!valid)
                 continue;
 
+            // don't spawn in AntiAnomalyZones
+            var antiAnomalyZonesQueue = AllEntityQuery<AntiAnomalyZoneComponent>();
+            while (antiAnomalyZonesQueue.MoveNext(out var uid, out var zone))
+            {
+                var zoneTile = _transform.GetGridTilePositionOrDefault(uid, gridComp);
+
+                var delta = (zoneTile - tile);
+                if (delta.Length < zone.ZoneRadius)
```

Comment:
```suggestion
                if (delta.LengthSquared() < zone.ZoneRadius * zone.ZoneRadius)
```
checking with length squared is more performant because it saves a square root check. not sure how pertinent it is here but this code sometimes runs a bunch of times in a row.